src/flows/context/flow.current.tsx

```diff
@@ -54,17 +67,62 @@ export const FlowProvider: FC = ({children}) => {
     }
   }
 
-  // NOTE this is a pretty awful mechanism, as it duplicates the source of
-  // truth for the definition of the current flow, but i can't see a good
-  // way around it. We need to ensure that we're still updating the values
-  // and references to the flows object directly to get around the async
-  // update issues.
+  const handleDeleteNotebook = useCallback(async () => {
+    event('delete_notebook')
+    try {
+      const response = await deleteNotebook({id})
+
+      if (response.status !== 204) {
+        throw new Error(response.data.message)
+      }
+      dispatch(notify(notebookDeleteSuccess()))
+    } catch (error) {
+      dispatch(notify(notebookDeleteFail()))
+    }
+  }, [dispatch, id])
+
+  const handleCloneNotebook = useCallback(async () => {
+    try {
+      const {flows} = await getAllAPI(orgID)
+
+      const allFlowNames = Object.values(flows).map(value => value.name)
+      const clonedName = incrementCloneName(allFlowNames, currentFlow.name)
+
+      const _flow = serialize({...currentFlow, name: clonedName})
+      delete _flow.data.id
+
+      const response = await postNotebook(_flow)
+
+      if (response.status !== 200) {
+        throw new Error(response.data.message)
+      }
+
+      return response.data.id
+    } catch (error) {
+      console.error({error})
+    }
+  }, [currentFlow, orgID])
+
+  const handleGetNotebook = useCallback(async notebookId => {
+    try {
```

Comment:
i think you have to add a loading state here now that it's being fetched from someplace else